### ISSUE TITLE
Kubernetes module with ConfigMap fails to report status due to missing serviceResource

### ISSUE THREAD

**Issue by ultimatemonty** [garden-io/garden#2476]
### Current Behavior
We have a kubernetes module that injects a ConfigMap into the kube-system namespace. This ConfigMap is adding some additional CoreDNS rules we need for our local development environment. When starting garden with `garden dev --force-refresh` we get the following error output:

```
Failed getting status for service 'kube-dns' (from module 'kube-dns'). Here is the output:
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
kubernetes module kube-dns doesn't specify a serviceResource in its configuration. You must specify a resource in the module config in order to use certain Garden features, such as
hot reloading, tasks and tests.
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
```

The ConfigMap still deploys properly but because the status isn't reported back properly any services that depend on it do not start.

### Expected behavior
the missing serviceResource is ignored for ConfigMap resources and they come up cleanly. Alternatively the feature request in #2456 would work

### Reproducible example
Here is our module config
```
apiVersion: garden.io/v0
kind: Module
type: kubernetes
name: kube-dns
description: Additional Core DNS rules for localdev
manifests:
  - apiVersion: v1
    kind: ConfigMap
    metadata:
      name: coredns
      namespace: kube-system
    data:
      Corefile: |
        .:53 {
          // CoreDNS rules here
        }
```

### Workaround
We're able to work around it by removing service dependencies on this module

### Suggested solution(s)
a ConfigMap specific module type for injecting additional configuration (ala what #2456 is proposing)

### Your environment

* **OS:** MacOS
* **How I'm running Kubernetes:** Docker for Mac

`garden version`
`0.12.23` (running latest master to test the fix for #2471)

**Comment by edvald** [comment]
This looks a lot like what #2471 fixed, but we may need to look at this as well. Workaround is to use other CLI commands basically, since this is an issue with the dev command.

**Comment by edvald** [comment]
I believe this is fixed by #2490.
